signer/core/signed_data_internal_test.go

```diff
@@ -17,10 +17,42 @@
 package core
 
 import (
+	"fmt"
 	"math/big"
 	"testing"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
 )
 
+func TestParseBytes(t *testing.T) {
+	for i, tt := range []struct {
+		v   interface{}
+		exp []byte
+	}{
+		{"0x", []byte{}},
+		{"0x1234", []byte{0x12, 0x34}},
+		{[]byte{12, 34}, []byte{12, 34}},
+		{hexutil.Bytes([]byte{12, 34}), []byte{12, 34}},
+		{"not a hex string", nil},
+		{15, nil},
+		{nil, nil},
+	} {
+		out, ok := parseBytes(tt.v)
+		if tt.exp == nil {
+			if ok {
+				t.Errorf("Case %d: expecting !ok, got ok with %v", i, out)
```

Comment:
Please print hexbytes
```suggestion
				t.Errorf("Case %d: expecting !ok, got ok with %x", i, out)
```

Reply:
Also, please add
```
			if len(out) != 0{
				t.Errorf("Case %d: expected zero output on failure, got %x", i, out)
			}
```